src/content/plugins.md

```diff
@@ -0,0 +1,325 @@
+---
+title: Broccoli Plugins
+---
+
+The first question you should ask yourself is "what is a Broccoli plugin?".
+
+Plugins are what a build pipeline developer will interact with most. Plugins are what do the actual work of
+transforming files at each step of build process. The API of a plugin requires just 2 steps, creating a class that
+extends the [broccoli-plugin](https://github.com/broccolijs/broccoli-plugin) base class, and implementing a 
+`build()` method that performs some work and/or returns a promise.
+
+${toc}
+
+Let's have a look at the basic building blocks of a Broccoli plugin.
+
+# Broccoli-Plugin base class
+
+```js
+import Plugin from 'broccoli-plugin';
+
+class MyPlugin extends Plugin {
+    constructor (inputNode, options) {
+        super([inputNodes], options);
+    }
+    
+    build() {
+        // A plugin can receive single or multiple inputs nodes/directories
+        // The files are available in this.inputPaths[0]/this.inputPaths[1]...
+        // You can do anything with these files that you can do in node
+        // The output of your plugin must write to this.outputPath
+    }
+}
+```
+
+## Constructor
+
+`inputNode`: This can be one of two things, a string or another Broccoli plugin. If a string is passed, Broccoli
+expects this to be a source directory and automatically converts this into a
+[broccoli-source](https://github.com/broccolijs/broccoli-source).
+
+When constructing your build pipeline you will be passing plugins as inputs to one another, these are commonly
+referred to as `inputNodes`. The `broccoli-plugin` base class expects an array of `inputNodes`, however the 
+majority of plugins will only require a single input, so feel free to pass an array up if only a single input is
```

Comment:
Suggestion:
so you can also pass in a single input if that’s all that is required

Reply:
I don't know if this is worth mentioning but even with this description of input nodes I still have the general confusion I mention [here](https://gist.github.com/machty/0360f5ef2a257dace0ddb5c1d153c963#array-of-inputnodes-vs-single-merged-inputnode): _when_ do I need more than one input node? What kinds of Broccoli plugins logically demand more than one node? If a node is essentially a collection of folders and files, and plugins can be past an array of collections, why not just a single collection?